ci/build-package-docs.sh

```diff
@@ -55,6 +55,9 @@ go get -u github.com/pkg/errors
 # Regenerate the Node.JS SDK docs
 PKGS=${PKG_NAME} NOBUILD=true ./scripts/run_typedoc.sh
 
+# Regenerate the resource docs (for the specific plugin version)
+./scripts/gen_resource_docs.sh ${PKG_NAME} true ${VERSION}
```

Comment:
Two things to note here. We should build an exclusion list here to exclude building the resource docs for the following `PKG_NAME` values:
* `pulumi`
* `random`
* `policy`
* `docker`
* `kubernetesx`
* `awsx`
* `eks`

And secondly, I think we should also clone `pulumi/pulumi` just like we do `pulumi/docs`, OR remove the local override from https://github.com/pulumi/docs/blob/master/tools/resourcedocsgen/go.mod. Otherwise, running this script will cause the `resourcedocsgen` tool to look for the `pulumi/pulumi/sdk` and `pulumi/pulumi/pkg` modules locally.

Personally, I prefer that we clone the `pulumi/pulumi` repo here. But I am not sure if there is a reason not to?

Reply:
> We should build an exclusion list here to exclude building the resource docs for the following `PKG_NAME` values

Ah, good point.

Reply:
> We should build an exclusion list here to exclude building the resource docs for the following `PKG_NAME` values:

So we skip calling `gen_resource_docs.sh` for those `${PKG_NAME}` values. Sure, I can do that.

But just for my own understanding, why do we want to do that? What is special about those packages that they shouldn't have resource docs generated for them?

>  I think we should also clone pulumi/pulumi just like we do pulumi/docs, OR remove the local override from .../pulumi/docs/.../resourcedocsgen/go.mod. 

I still ramping up on how all of the repos fit together for generating the resource docs. But if I understand correctly, the `gen_resource_docs.sh` script assumes that the `pulumi/pulumi` repo exists on the local machine. It has the following snippet inside of it:

```
echo "Generating docs templates bundle in pulumi/pulumi"
pushd ../pulumi
make generate
popd
```

So you are suggesting that we run `git clone pulumi/pulumi` from within `ci/build-package-docs.sh`? (Since if `gen_resource_docs.sh` is ran for a provider repo that _doesn't_ clone `pulumi/pulumi`, then calling `gen_resource_docs.sh` would fail.) So far so good?

I'll look into doing that, too.

Reply:
> But just for my own understanding, why do we want to do that? What is special about those packages that they shouldn't have resource docs generated for them?

Sorry, I should have mentioned the _why_ earlier in my comment, but it is because we don't produce a schema for those packages. So we can't generate resource docs for them.

> So you are suggesting that we run git clone pulumi/pulumi from within ci/build-package-docs.sh? (Since if gen_resource_docs.sh is ran for a provider repo that doesn't clone pulumi/pulumi, then calling gen_resource_docs.sh would fail.) So far so good?

Correct. For generating a static bundle of the templates, as well as to actually run the docs generator itself (and this is where that go.mod file comes into play).

Reply:
> But just for my own understanding, why do we want to do that? What is special about those packages that they shouldn't have resource docs generated for them?

Some of these don't actually have any resources like `pulumi` and `policy` and others are higher-level libraries with component resources (e.g. `kubernetesx`, `awsx`, and `eks`) that haven't been "schematized" yet and therefore don't have resource-oriented docs yet.

@praneetloke, but I am confused about `random` and `docker`. Why should those be excluded?

Reply:
> but I am confused about random and docker. Why should those be excluded?

Ha, I assumed `random` and `docker` are in the same bucket. But you are right, they should not be excluded. In fact, these two providers seem to have been updated already to support generating a schema.